plugins/woocommerce/src/Admin/API/MarketingCampaigns.php

```diff
@@ -132,6 +132,57 @@ public function get_items( $request ) {
 		return $response;
 	}
 
+	/**
+	 * Get formatted price based on Price type.
+	 *
+	 * This uses plugins/woocommerce/i18n/currency-info.php and plugins/woocommerce/i18n/locale-info.php to get option object based on $price->currency.
+	 *
+	 * Example:
+	 *
+	 * - When $price->currency is 'USD' and $price->value is '1000', it should return '$1000.00'.
+	 * - When $price->currency is 'JPY' and $price->value is '1000', it should return '¥1,000'.
+	 * - When $price->currency is 'AED' and $price->value is '1000', it should return '5.000,00 د.إ'.
+	 *
+	 * @param Price $price Price object.
+	 * @return String formatted price.
+	 */
+	private function get_formatted_price( $price ) {
+		// Get $currency_info based on user locale or default locale.
+		$user_locale = get_user_locale();
+		$currency_info_all = include WC()->plugin_path() . '/i18n/currency-info.php';
+		$currency_info_locales = $currency_info_all[ $price->get_currency() ];
+		$currency_info     = $currency_info_locales[ $user_locale ] ?? $currency_info_locales['default'];
+
+		// Get $num_decimals to be passed to wc_price.
+		$locale_info_all = include WC()->plugin_path() . '/i18n/locale-info.php';
+		$locale_index    = array_search( $price->get_currency(), array_column( $locale_info_all, 'currency_code' ), true );
+		$num_decimals    = array_values( $locale_info_all )[ $locale_index ]['num_decimals'];
```

Comment:
```suggestion
		// Get $num_decimals and $currency_info to be passed to wc_price.
		$locale_info_all  = include WC()->plugin_path() . '/i18n/locale-info.php';
		$locale_index     = array_search( $price->get_currency(), array_column( $locale_info_all, 'currency_code' ), true );
		$locale           = array_values( $locale_info_all )[ $locale_index ];
		$currency_locales = $locale['locales'];
		$user_locale      = get_user_locale();
		$currency_info    = $currency_locales[ $user_locale ] ?? $currency_locales['default'];
		$num_decimals     = $locale['num_decimals'];
		
```

locale-info.php already has currency-info.php values embedded in the structure so there is no need to load it twice.

Reply:
Thanks, this looks good! Done in https://github.com/woocommerce/woocommerce/pull/44917/commits/9f1d749df5f7292aea1dc99ae0eb12b86fdafb62.